src/Name/FunctionName.php

```diff
@@ -76,7 +76,7 @@ public function equals(FunctionName $name)
      * Checks the name of a function and throw an exception if is not valid.
      * When name is valid returns an array of the name itself and its namespace parts.
      *
-     * @param  string $function_name
+     * @param  string|null $function_name
```

Comment:
"In theory" this should always be a string. Unfortunately not being PHP 7+, I can't use scalar type declaration in the constructor, so I can say that I "expect" a string, but can't be sure.

So if declare doc bloc based on what I expect this should be just `string`, if I declare based on what it could actually be, then it should be `mixed` because people can pass what they want in constructor.

So `string|null` is quite arbitrary...